.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,49 @@
+* Create tutorials, blog posts and example use-cases on how to use Goss.
+* Help users with [questions](https://github.com/aelsabbahy/goss/labels/question) tracker.
+* Fix verified [bugs](https://github.com/aelsabbahy/goss/labels/bug).
+* Implement approved [feature requests](https://github.com/aelsabbahy/goss/labels/feature).
+* Spread the word.
+
+
+Features and bug reports and questions
+--------------------------------------
+
+Please search the [issues](https://github.com/aelsabbahy/goss/issues) page before opening a feature request or a bug report. If a feature or a bug report already exists, please thumbs up the initial post to indicate it's importance to you and raise it's priority. Please comment and contribute to said issue if you feel it's deficient.
+
+Bug reports
+-----------
+
+If you think you found a bug in Goss, please submit a [bug report](https://github.com/aelsabbahy/goss/issues).
+
+Feature requests
+----------------
+
+If there's a feature you wish Goss would support, please open a feature request.
+
+Some things to note prior to opening a Goss feature request:
+* Goss is intended to be quick and easy to learn.
+* Goss is focused on the 20% of the 80/20 rule. In other words, Goss focuses on the 20% of features that cover the core aspects of OS testing and benefit 80% of users.
+* Goss provides a generic [command](https://github.com/aelsabbahy/goss/blob/master/docs/manual.md#command) runner to allow users to cover more nuanced test cases.
+
+If you believe your feature adheres to the goals of Goss, please open a [feature request](https://github.com/aelsabbahy/goss/issues) on GitHub which describes the feature you would like to see, why it is useful, and how it should work.
+
+Once a feature is submitted, it will be reviewed. Upon approval, the issue can be worked on and PRs can be submitted that implement this new feature.
+
+
+Contributing code and documentation changes
+-------------------------------------------
+
+If you have a bugfix or new feature that you would like to contribute to Goss, please find or open an issue about it first. Talk about what you would like to do. It may be that somebody is already working on it, or that there are particular issues that you should know about before implementing the change.
+
+We enjoy working with contributors to get their code accepted. There are many approaches to fixing a problem and it is important to find the best approach before writing too much code.
+
+Note that it is unlikely the project will merge refactors for the sake of refactoring or niche features that aren't common use-cases (see the feature request section above). These types of pull requests have a high cost to maintainers but provide little benefit to the community.
+
+Lastly, in order for a pull request to be merged, it must provide tests proving the change works as intended, this also prevents future changes from introducing regressions.
```

Comment:
```suggestion
Lastly, in order for a pull request to be merged, it must provide automated tests (unit and integration) proving the change works as intended, this also prevents future changes from introducing regressions. It would be quite odd for a testing tool to not have a healthy approach to test automation, after all. :-)
```

Reply:
> Goss currently is lacking in unit tests (shame on me), feels a big hypocritical to require it for all future work.

Not really; sometimes the value of adding tests isn't high enough compared to the effort needed.

Reading your response, though; it occurs to me that another way a person might contribute is by adding unit/integration test coverage where currently there are gaps. Maybe add that?

One thing we could do to encourage contributors is create issues for not just features but items of non-functional requirements like test coverage or improving processes; what do you think?

Reply:
Maybe something along the lines of "unit and/or integration tests as applicable to the feature under testing". Then you could add a section going into more detail about what tests make sense, and maybe even the styling of testing.

I like @petemounce suggestion of issues/PRs to increasing or refactoring the test suite.